### PR TITLE
HAProxy: update the recommended hardware configuration for HAProxy

### PR DESCRIPTION
**Pull request by djshow832**
### What is changed, added or deleted? (Required)

1. The least hardware configuration for HAProxy was too low, making it almost impossible to work for users. I've added the highest QPS for this configuration so that users can estimate the configuration for their workload.
2. The recommended memory is inconsistent with HAProxy official docs.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.6 (TiDB 7.6 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [x] v7.4 (TiDB 7.4 versions)
- [x] v7.3 (TiDB 7.3 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: best-practices/haproxy-best-practices.md [hunk 1]
```diff
@@ -34,12 +34,12 @@ HAProxy 由 Linux 内核的核心贡献者 Willy Tarreau 于 2000 年编写，
 
 ### 硬件要求
 
-根据官方文档，对 HAProxy 的服务器硬件配置有以下建议，也可以根据负载均衡环境进行推算，在此基础上提高服务器配置。
+根据[官方文档](https://www.haproxy.com/documentation/haproxy-enterprise/getting-started/installation/linux/)，HAProxy 的服务器硬件的最低配置如下。在 sysbench oltp_read_write 工作负载下，该配置的最高 QPS 约为 50K。你可以根据负载均衡环境进行推算，在此基础上提高服务器配置。
```
Comment (hfxsd) [inline best-practices/haproxy-best-practices.md:37]:
```suggestion
根据[官方文档](https://www.haproxy.com/documentation/haproxy-enterprise/getting-started/installation/linux/)，HAProxy 的服务器硬件的最低配置如下。在 sysbench `oltp_read_write` 工作负载下，该配置的最高 QPS 约为 50K。你可以根据负载均衡环境进行推算，在此基础上提高服务器配置。
```

**Review by xhebox — APPROVED** [review]

**Review by hfxsd — APPROVED** [review]

File: best-practices/haproxy-best-practices.md [hunk 2]
```diff
@@ -34,12 +34,12 @@ HAProxy 由 Linux 内核的核心贡献者 Willy Tarreau 于 2000 年编写，
 
 ### 硬件要求
 
-根据官方文档，对 HAProxy 的服务器硬件配置有以下建议，也可以根据负载均衡环境进行推算，在此基础上提高服务器配置。
+根据[官方文档](https://www.haproxy.com/documentation/haproxy-enterprise/getting-started/installation/linux/)，HAProxy 的服务器硬件的最低配置如下。在 sysbench `oltp_read_write` 工作负载下，该配置的最高 QPS 约为 50K。你可以根据负载均衡环境进行推算，在此基础上提高服务器配置。
```
Comment (qiancai) [inline best-practices/haproxy-best-practices.md:37]:
```suggestion
根据[HAProxy 官方文档](https://www.haproxy.com/documentation/haproxy-enterprise/getting-started/installation/linux/)，HAProxy 的服务器硬件的最低配置如下。在 Sysbench `oltp_read_write` 工作负载下，该配置的最高 QPS 约为 50K。你可以根据负载均衡环境进行推算，在此基础上提高服务器配置。
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

**Pull request merged**